src/Symfony/Component/Cache/Adapter/ArrayAdapter.php

```diff
@@ -0,0 +1,180 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Cache\Adapter;
+
+use Psr\Cache\CacheItemInterface;
+use Psr\Cache\CacheItemPoolInterface;
+use Symfony\Component\Cache\CacheItem;
+use Symfony\Component\Cache\Exception\InvalidArgumentException;
+
+/**
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class ArrayAdapter implements CacheItemPoolInterface
```

Comment:
@andrerom Your concern is filling memory with stale data or something more? There is no risk to return stale data, since the expiry is verified. The use case I see is when a non persistent in memory cache is enough. Could be testing or something else. We could enhance the logic for dealing with expirations, such has maintaining a property that stores the next full scan for expired items, to be checked e.g. on every access. Would you like to work on it?